Subpath exports missing 'types' condition for moduleResolution: node

## Description

The `docxodus/react` subpath export works at runtime via bundlers (Vite, webpack) that understand `package.json` `"exports"`, but fails TypeScript compilation when `tsconfig.json` uses `"moduleResolution": "node"` (the most common setting in existing projects).

## Error

```
error TS2307: Cannot find module 'docxodus/react' or its corresponding type declarations.
  There are types at '/path/to/node_modules/docxodus/dist/react.d.ts', but this result
  could not be resolved under your current 'moduleResolution' setting. Consider updating
  to 'node16', 'nodenext', or 'bundler'.
```

## Root Cause

In `package.json`, the `"./react"` export only has `"import"` and `"types"` conditions:

```json
{
  "exports": {
    "./react": {
      "import": "./dist/react.js",
      "types": "./dist/react.d.ts"
    }
  }
}
```

With `"moduleResolution": "node"`, TypeScript ignores the `"exports"` field entirely and falls back to looking for `docxodus/react.d.ts` or `docxodus/react/index.d.ts` at the package root — neither exists.

## Suggested Fix

Add a `typesVersions` field to `package.json`, which is respected by all `moduleResolution` modes:

```json
{
  "typesVersions": {
    "*": {
      "react": ["dist/react.d.ts"],
      "worker": ["dist/worker.d.ts"],
      "pagination": ["dist/pagination.d.ts"]
    }
  }
}
```

This makes `import { PaginatedDocument } from "docxodus/react"` resolve types correctly regardless of `moduleResolution` setting.

## Current Workaround

```typescript
// @ts-expect-error -- tsconfig uses moduleResolution:"node" which doesn't
// support package.json "exports" subpaths. The import works at runtime via Vite.
import { PaginatedDocument } from "docxodus/react";
```

## Environment

- docxodus: 5.5.2
- TypeScript: 5.x
- tsconfig `moduleResolution`: `"node"`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,9 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased] - .NET 8 / Open XML SDK 3.x Migration
 
+### Fixed (npm)
+- **TypeScript subpath exports not resolving under `moduleResolution: "node"` (Issue #113)** - Added `typesVersions` fallback to npm package.json so `docxodus/react` and `docxodus/worker` subpath imports resolve types correctly under all TypeScript module resolution modes. Also reordered export conditions to put `types` before `import` per TypeScript requirements.
+
 ### Added
 - **Incremental annotation overlay API (Issue #106)** - Decouple HTML conversion from annotation projection to avoid full WASM re-conversion
   - `ProjectAnnotationsOntoHtml()` - Project a full annotation set onto already-converted HTML

--- a/npm/package.json
+++ b/npm/package.json
@@ -8,16 +8,22 @@
   "types": "./dist/index.d.ts",
   "exports": {
     ".": {
-      "import": "./dist/index.js",
-      "types": "./dist/index.d.ts"
+      "types": "./dist/index.d.ts",
+      "import": "./dist/index.js"
     },
     "./react": {
-      "import": "./dist/react.js",
-      "types": "./dist/react.d.ts"
+      "types": "./dist/react.d.ts",
+      "import": "./dist/react.js"
     },
     "./worker": {
-      "import": "./dist/worker-proxy.js",
-      "types": "./dist/worker-proxy.d.ts"
+      "types": "./dist/worker-proxy.d.ts",
+      "import": "./dist/worker-proxy.js"
+    }
+  },
+  "typesVersions": {
+    "*": {
+      "react": ["./dist/react.d.ts"],
+      "worker": ["./dist/worker-proxy.d.ts"]
     }
   },
   "files": [

